src/ShallowTraversal.js

```diff
@@ -66,6 +69,18 @@ export function nodeHasId(node, id) {
   return propsOfNode(node).id === id;
 }
 
+
+export function nodeHasProperty(node, propKey, propValue) {
+  const nodeProps = propsOfNode(node);
+
+  if (propValue) {
+    return nodeProps[propKey] === propValue;
```

Comment:
yah... so we may need some special case logic put in here. let's try and figure out what the corner cases are and what the desired behavior is...